lib/_http_outgoing.js

```diff
@@ -337,6 +337,14 @@ OutgoingMessage.prototype.getHeader = function(name) {
 };
 
 
+OutgoingMessage.prototype.getAllHeaders = function() {
+  if (!this._headers)
+    return;
+  else
+    return util._extend({}, this._headers);
```

Comment:
> The performance hit is unfortunate. Anyone know if v8 is working on making that better?

No, V8 really wants you to use `Map` for these kind of use cases, and `"use strong"` even makes it impossible to abuse objects as maps.